ha_lineairdb.cc

```diff
@@ -848,6 +869,56 @@ void ha_lineairdb::set_write_buffer() {
 
 bool ha_lineairdb::is_primary_key_exists() { return (0 < table->s->keys); }
 
+bool ha_lineairdb::store_blob_to_field(Field** field) {
+  if ((*field)->is_flag_set(BLOB_FLAG)) {
+    Field_blob* blob_field = down_cast<Field_blob*>(*field);
+    size_t length          = blob_field->get_length();
+    if (length > 0) {
+      unsigned char* new_blob = new (&blobroot) unsigned char[length];
+      if (new_blob == nullptr) return true;
+      memcpy(new_blob, blob_field->get_blob_data(), length);
+      blob_field->set_ptr(length, new_blob);
+    }
+  }
+  return false;
+}
+
+bool ha_lineairdb::is_over_buf_flag_capacity(int nullable_field_index) {
+  return nullable_field_index == BYTE_BIT_NUMBER;
+}
+
+void ha_lineairdb::flush_null_flag_to_buf(uchar* buf, std::bitset<BYTE_BIT_NUMBER> &nullBit, 
+                                          int &nullable_field_index, int &buf_nullbyte_index) {
+  uchar mask = nullBit.to_ulong();
+  memcpy(&buf[buf_nullbyte_index], &mask, 1);
+  buf_nullbyte_index++;
+  nullable_field_index = 0;
+  nullBit.set();
+}
+
+bool ha_lineairdb::set_flag_for_nonnull_field(std::bitset<BYTE_BIT_NUMBER> &nullBit, int &nullable_field_index) {
+  bool field_is_null = true;
+  if (write_buffer_.length() != 4 || strncmp(write_buffer_.c_ptr(), "NULL", 4)) { 
+    nullBit.flip(nullable_field_index); 
+    field_is_null = false;
+  }
+  return field_is_null;
+}
+
+bool ha_lineairdb::handle_null_field(uchar* buf, Field** field,
+                                    std::bitset<BYTE_BIT_NUMBER> &nullBit, 
+                                    int &nullable_field_index, int &buf_nullbyte_index) {
```

Comment:
Such a large number of arguments (of complex types) passed to the functions means that these functions are in fact no more than a single procedure divided up.

What should be separated is not a procedure, but a reusable process.

Reply:
I thought `set_fields_from_lineairdb` would be more readable after abstracting `handle_null_field` out.
Is it bad practice to separate procedures like this?

Reply:
I don't think it's a bad practice, but I think this just shortens the function lines.
I was a little painful to read.
I think it could be better with a few things.

1. Naming: In `handle_null_field`, I didn't know what `handle` meant. I think it can be improved.
2. Too many complicated arguments: I think it expects to inherit the almost context of the caller. It means just jumping.

I think the best practice is, to define a function in a manner that makes it easy to understand what the function does by looking at the name of the function and its arguments.

Reply:
I see, thank you for the advice.
I think handle_null_field function cannot have a name more specific so I will put it back in `set_fields_from_lineairdb`.